default.conf

```diff
@@ -11,9 +11,20 @@ server {
     resolver $RESOLVERS valid=10s ipv6=off;
     set $upstream $UPSTREAM_HTTP_ADDRESS;
 
+    # Extract hostname from the upstream URL to use in the Host header
+    set $upstream_host $upstream;
+    if ($upstream ~* ^https?://([^:/]+)) {
+        set $upstream_host $1;
+    }
+
+    # Add these SSL configuration settings, in case upstream is a HTTPS server
+    proxy_ssl_server_name on;
+    proxy_ssl_protocols TLSv1.2 TLSv1.3;
+    proxy_ssl_verify off;
+
     location / {
         proxy_pass $upstream;
-        proxy_set_header Host $http_host;
```

Comment:
The point of this line was to ensure we pass the same HOST to the upstream. With this change, we're always passing the HOST of the upstream itself which breaks the existing applications. 

I think we may be able to do this, but we need to limit this only to HTTPS upstream. If the upstream is a plain HTTP, we need to ensure we're not changing this behavior.